src/renderers/shared/fiber/__tests__/ReactIncrementalSideEffects-test.js

```diff
@@ -127,6 +127,66 @@ describe('ReactIncremental', function() {
 
   });
 
+  it('can reuse side-effects after being preempted', function() {
+
+    function Bar(props) {
+      return <span prop={props.children} />;
+    }
+
+    var middleContent = (
+      <div>
+        <Bar>Hello</Bar>
+        <Bar>World</Bar>
+      </div>
+    );
+
+    function Foo(props) {
+      return (
+        <div hidden={true}>
+          {
+            props.step === 0 ?
+              <div>
+                <Bar>Hi</Bar>
+                <Bar>{props.text}</Bar>
+              </div>
+              : middleContent
+          }
+        </div>
+      );
+    }
+
+    // Init
+    ReactNoop.render(<Foo text="foo" step={0} />);
+    ReactNoop.flush();
+
+    expect(ReactNoop.root.children).toEqual([
+      div(div(span('Hi'), span('foo'))),
+    ]);
+
+    // Make a quick update which will schedule low priority work to
+    // update the middle content.
+    ReactNoop.render(<Foo text="bar" step={1} />);
+    ReactNoop.flushLowPri(30);
+
+    // The tree remains unchanged.
+    expect(ReactNoop.root.children).toEqual([
+      div(div(span('Hi'), span('foo'))),
+    ]);
+
+    // The first Bar has already completed its update but we'll interupt it to
```

Comment:
Just because of the fixed time elapsed in the unit test. The unit tests rely on implementation details taking a certain time. This is verified in the parallel test in ReactIncremental-test. I was thinking about testing this here too but the test became so complicated. I guess it is not ideal that the timing aspect here is a bit of a fragile test.